discovery_engine/test/integration/migration_test.dart

```diff
@@ -0,0 +1,362 @@
+        image: Uri.parse('http://$source4/foo/imag.png'),
+        rank: 4,
+        score: 4,
+        snippet: 'snippet4',
+        sourceDomain: source4,
+        title: 'title4',
+        topic: 'topic4',
+        url: Uri.parse('http://$source4/foo/index.html'),
+      ),
+    )
+  ];
+
+  group('Hive to Rust migration', () {
+    late LocalNewsApiServer server;
+    late TestEngineData data;
+    DiscoveryEngine? engine;
+
+    setUp(() async {
+      data = await setupTestEngineData(useEphemeralDb: false);
+      server = await LocalNewsApiServer.start();
+
+      EventHandler.registerHiveAdapters();
+      await EventHandler.initDatabase(data.applicationDirectoryPath);
+    });
+
+    tearDown(() async {
+      await engine?.dispose();
+      await server.close();
+      await Directory(data.applicationDirectoryPath).delete(recursive: true);
+    });
+
+    test('works with mostly empty db (a)', () async {
+      await HiveInjection.accessRepositories(data.applicationDirectoryPath,
+          (repos) async {
+        await repos.sourceReactedRepository.save(
+          SourceReacted(source1, true),
+        );
+      });
+
+      engine = await initEngine(data, server.port);
+
+      await HiveInjection.accessRepositories(data.applicationDirectoryPath,
+          (repos) async {
+        expect(repos.engineStateRepository.isEmpty, isTrue);
+        expect(repos.documentRepository.isEmpty, isTrue);
+        expect(repos.activeSearchRepository.isEmpty, isTrue);
+        expect(repos.activeDocumentDataRepository.isEmpty, isTrue);
+        expect(repos.sourceReactedRepository.isEmpty, isTrue);
+        expect(repos.sourcePreferenceRepository.isEmpty, isTrue);
+      });
+    });
+
+    test('works with mostly empty db (b)', () async {
+      await HiveInjection.accessRepositories(data.applicationDirectoryPath,
+          (repos) async {
+        await repos.documentRepository.update(docs[0]);
+      });
+
+      engine = await initEngine(data, server.port);
+
+      final feed =
+          expectEvent<RestoreFeedSucceeded>(await engine!.restoreFeed()).items;
+      expect(feed[0], equals(docs[0].toApiRepr()));
+      expect(feed.length, equals(1));
+
+      await HiveInjection.accessRepositories(data.applicationDirectoryPath,
+          (repos) async {
+        expect(repos.engineStateRepository.isEmpty, isTrue);
+        expect(repos.documentRepository.isEmpty, isTrue);
+        expect(repos.activeSearchRepository.isEmpty, isTrue);
+        expect(repos.activeDocumentDataRepository.isEmpty, isTrue);
+        expect(repos.sourceReactedRepository.isEmpty, isTrue);
+        expect(repos.sourcePreferenceRepository.isEmpty, isTrue);
+      });
+    });
+
+    test('works with needing migration for all data', () async {
+      await HiveInjection.accessRepositories(data.applicationDirectoryPath,
+          (repos) async {
+        //FIXME test engine state migration by having a known valid engine state blob
```

Comment:
that's a good idea, I think I have some hive files with some engine state that we could use from the datetime type migration issue, or if you need a particular state, then I can help also to create it